submission/submission-webapp/src/test/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmissionWebHandlerTest.kt

```diff
@@ -0,0 +1,119 @@
+import ebi.ac.uk.api.dto.SubmissionDto
+import ebi.ac.uk.dsl.submission
+import ebi.ac.uk.io.sources.ComposedFileSource
+import ebi.ac.uk.security.integration.model.api.SecurityUser
+import io.github.glytching.junit.extension.folder.TemporaryFolder
+import io.github.glytching.junit.extension.folder.TemporaryFolderExtension
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.impl.annotations.SpyK
+import io.mockk.junit5.MockKExtension
+import io.mockk.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.web.multipart.MultipartFile
+
+@ExtendWith(TemporaryFolderExtension::class, MockKExtension::class)
+class SubmissionWebHandlerTest(
+    temporaryFolder: TemporaryFolder,
+    @MockK private val user: SecurityUser,
+    @MockK private val pageTabReader: PageTabReader,
+    @MockK private val multiPartAttachedFile: MultipartFile,
+    @MockK private val submissionService: SubmissionService,
+    @MockK private val tempFileGenerator: TempFileGenerator,
+    @MockK private val multiPartSubmissionFile: MultipartFile,
+    @MockK private val composedFileSource: ComposedFileSource,
+    @MockK private val serializationService: SerializationService
+) {
+    private val submissionFilter = SubmissionFilter()
+    private val submission = submission("S-TEST123") { }
+    private val submissionFile = temporaryFolder.createFile("submission.tsv")
+    private val submissionDB = SubmissionDB("S-TEST123").apply {
+        title = "Test Submission"
+        releaseTime = 123L
+        creationTime = 123L
+        modificationTime = 123L
+    }
+
+    @SpyK private var testInstance =
+        SubmissionWebHandler(pageTabReader, submissionService, tempFileGenerator, serializationService)
+
+    @BeforeEach
+    fun beforeEach() {
+        every { pageTabReader.read(submissionFile) } returns "Submission"
+        every { serializationService.getSubmissionFormat(submissionFile) } returns TSV
+        every { tempFileGenerator.asFile(multiPartSubmissionFile) } returns submissionFile
+        every { submissionService.deleteSubmission("S-TEST123", user) } answers { nothing }
+        every { submissionService.submit(submission, user, composedFileSource) } returns submission
+        every { testInstance.getUserFilesSource(user, "Submission", TSV) } returns composedFileSource
+        every { submissionService.getSubmissions(user, submissionFilter) } returns listOf(submissionDB)
+        every { serializationService.deserializeSubmission("Submission", TSV, composedFileSource) } returns submission
+        every {
+            testInstance.getComposedFilesSource(user, arrayOf(multiPartAttachedFile), "Submission", TSV)
+        } returns composedFileSource
```

Comment:
this one may become hard to maintain as it is hard to track what mock is used for which test, for example, I can imagine ourselves removing a method and forgetting to remove mock. So my suggestion is to group test that needs the same setup using  `@Nested`  and move the specific mock setup to the test when is actually need it.